app/redux/actions/async.js

```diff
@@ -1169,7 +1170,10 @@ export function fetchPatientData(api, options, id) {
 
       // We only add the data to the worker if another patient id has not been fetched
       // while we waited on this one, and we are still on an app view specific to that patient
-      if (location.pathname.indexOf(id) >= 0 && (!fetchingPatientId || fetchingPatientId === id)) {
+      if (
+        options.forceDataWorkerAddDataRequest ||
+        (location.pathname.indexOf(id) >= 0 && (!fetchingPatientId || fetchingPatientId === id))
+      ) {
```

Comment:
Here, I force `actions.async.fetchPatientData()` to call `actions.worker.dataWorkerAddDataRequest()`.

I didn't want to do it this way, I would have preferred to call `actions.worker.dataWorkerAddDataRequest()` from my component after we verify that the patient is loaded. However, the call to `dataWorkerAddDataRequest()` requires `options.startDate` as an arg, and the startDate is generated in a complex fashion within this function. I couldn't come up with a better way.